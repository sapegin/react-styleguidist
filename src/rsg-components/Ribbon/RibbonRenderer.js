import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';

export const styles = ({ color, fontSize, fontFamily }) => ({
	root: {
		position: 'fixed',
		top: 0,
		right: 0,
		width: '149px',
		height: '149px',
		zIndex: 999,
	},
	link: {
		fontFamily: fontFamily.base,
		position: 'relative',
		right: -37,
		top: -22,
		display: 'block',
		width: 190,
		padding: '5px 15px',
		textAlign: 'center',
		color: 'white',
		fontSize: fontSize.base,
		background: color.ribbonBackground,
		textDecoration: 'none',
		textShadow: '0 -1px 0 rgba(0,0,0,.15)',
		transformOrigin: '0 0',
		transform: 'rotate(45deg)',
	},
});

export function RibbonRenderer({ classes, url, text, color, background }) {
	return (
		<div className={classes.root}>
			<a href={url} className={classes.link} style={{ color, background }}>
				{text || 'Fork me on GitHub'}
			</a>
		</div>
	);
}

RibbonRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	url: PropTypes.string.isRequired,
	text: PropTypes.string,
	color: PropTypes.string,
	background: PropTypes.string,
};

export default Styled(styles)(RibbonRenderer);
