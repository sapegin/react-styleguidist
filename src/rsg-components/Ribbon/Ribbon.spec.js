import React from 'react';
import { RibbonRenderer, styles } from './RibbonRenderer';

const props = {
	classes: classes(styles),
};

it('should render ribbon with url', () => {
	const actual = shallow(<RibbonRenderer {...props} url="http://example.com" />);

	expect(actual).toMatchSnapshot();
});

it('should render ribbon with a text', () => {
	const actual = shallow(
		<RibbonRenderer {...props} url="http://example.com" text="Share the repo" />
	);

	expect(actual).toMatchSnapshot();
});

it('should render ribbon with custom background and color', () => {
	const actual = shallow(
		<RibbonRenderer
			{...props}
			url="http://example.com"
			text="Share the repo"
			background="#cdcdcd"
			color="#aa00bb"
		/>
	);

	expect(actual).toMatchSnapshot();
});
