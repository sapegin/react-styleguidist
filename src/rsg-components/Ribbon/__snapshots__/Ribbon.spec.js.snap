// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ribbon with a text 1`] = `
<div
  className="root"
>
  <a
    className="link"
    href="http://example.com"
    style={
      Object {
        "background": undefined,
        "color": undefined,
      }
    }
  >
    Share the repo
  </a>
</div>
`;

exports[`should render ribbon with custom background and color 1`] = `
<div
  className="root"
>
  <a
    className="link"
    href="http://example.com"
    style={
      Object {
        "background": "#cdcdcd",
        "color": "#aa00bb",
      }
    }
  >
    Share the repo
  </a>
</div>
`;

exports[`should render ribbon with url 1`] = `
<div
  className="root"
>
  <a
    className="link"
    href="http://example.com"
    style={
      Object {
        "background": undefined,
        "color": undefined,
      }
    }
  >
    Fork me on GitHub
  </a>
</div>
`;
